sdk/packages/selector-solana/src/adapter.ts

```diff
@@ -0,0 +1,375 @@
+  AppInitData,
+  clearRecentStandardWalletForNetwork,
+  clearSessionIdForNetwork,
+  getRecentStandardWalletForNetwork,
+  isDesktopConnectedForNetwork,
+  isMobileBrowser,
+  IWalletListItem,
+  logoBase64,
+  NightlyConnectSelectorModal,
+  persistDesktopConnectForNetwork,
+  persistRecentStandardWalletForNetwork,
+  triggerConnect
+} from '@nightlylabs/wallet-selector-base'
+import {
+  BaseMessageSignerWalletAdapter,
+  WalletAdapterCompatibleStandardWallet,
+  WalletName,
+  WalletNotConnectedError,
+  WalletNotReadyError,
+  WalletReadyState,
+  WalletSignTransactionError,
+  isVersionedTransaction
+} from '@solana/wallet-adapter-base'
+import { StandardWalletAdapter } from '@solana/wallet-standard'
+import { PublicKey, Transaction, TransactionVersion, VersionedTransaction } from '@solana/web3.js'
+
+export class NightlyConnectAdapter extends BaseMessageSignerWalletAdapter {
+  name = 'Nightly Connect' as WalletName<'Nightly Connect'>
+  url = 'https://nightly.app'
+  icon = logoBase64
+
+  readonly supportedTransactionVersions: ReadonlySet<TransactionVersion> = new Set(['legacy', 0])
+
+  private _connecting: boolean
+  private _connected: boolean
+  private _publicKey: PublicKey | null
+  private _readyState: WalletReadyState =
+    typeof window === 'undefined' || typeof document === 'undefined'
+      ? WalletReadyState.Unsupported
+      : WalletReadyState.NotDetected
+
+  private _app: AppSolana | undefined
+  private _appSessionActive: boolean
+  private _innerStandardAdapter: StandardWalletAdapter | undefined
+  private _modal: NightlyConnectSelectorModal | undefined
+
+  private _appInitData: AppInitData
+  private _eagerConnectForStandardWallets: boolean
+
+  private _walletsList: IWalletListItem[] = []
+
+  private _chosenMobileWalletName: string | undefined
+
+  constructor(appInitData: AppInitData, eagerConnectForStandardWallets?: boolean) {
+    super()
+    this._connecting = false
+    this._connected = false
+    this._publicKey = null
+    this._appInitData = appInitData
+    this._eagerConnectForStandardWallets = !!eagerConnectForStandardWallets
+    this._appSessionActive = false
+  }
+
+  get connecting() {
+    return this._connecting
+  }
+
+  get connected() {
+    return this._connected
+  }
+
+  get readyState() {
+    return this._readyState
+  }
+
+  get publicKey() {
+    return this._publicKey
+  }
+
+  public static async build(
+    appInitData: AppInitData,
+    eagerConnectForStandardWallets?: boolean,
+    anchorRef?: HTMLElement,
+    onOpen?: () => void,
+    onClose?: () => void
+  ) {
+    const adapter = new NightlyConnectAdapter(appInitData, eagerConnectForStandardWallets)
+
+    await adapter.eagerConnectToRecentStandardWallet()
+
+    adapter._readyState = WalletReadyState.Installed
+
+    return adapter
+  }
+
+  public static buildLazy(
+    appInitData: AppInitData,
+    eagerConnectForStandardWallets?: boolean,
+    anchorRef?: HTMLElement,
+    onOpen?: () => void,
+    onClose?: () => void
```

Comment:
this is not necessary you should use events instead of callbacks

Reply:
removed